File: Frontend/src/app/modules/admin/services/admin-service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage-service';

const BASE_URL = "http://localhost:8080/"

@Injectable({
  providedIn: 'root'
})
export class AdminService {

  constructor(private http: HttpClient) { }

  getAllCars(): Observable<any> {
    return this.http.get(BASE_URL + "api/admin/cars", {
      headers: this.createAuthorizationHeader()
    });
  }

  getBids(): Observable<any> {
    return this.http.get(BASE_URL + "api/admin/car/bids", {
      headers: this.createAuthorizationHeader()
    });
  }

  updateBidStatus(bidId: number, status: string): Observable<any> {
    return this.http.get(BASE_URL + `api/admin/car/bids/${bidId}/${status}`, {
      headers: this.createAuthorizationHeader()
    });
  }

  searchCar(searchDto: any): Observable<any> {
    return this.http.post(BASE_URL + "api/admin/car/search", searchDto, {
      headers: this.createAuthorizationHeader()
    });
  }

  createAuthorizationHeader(): HttpHeaders {
    let authHeaders: HttpHeaders = new HttpHeaders();
    return authHeaders.set(
      'Authorization',
      'Bearer ' + StorageService.getToken()
    );
  }


}

File: Backend/src/main/java/com/luxorive/backend/services/admin/AdminService.java
package com.luxorive.backend.services.admin;

import com.luxorive.backend.dto.BidDTO;
import com.luxorive.backend.dto.CarDTO;
import com.luxorive.backend.dto.SearchCarDTO;

import java.util.List;

public interface AdminService {

    List<CarDTO> getAllCars();

    CarDTO getCarById(Long Id);

    void deleteCar(Long Id);

    List<CarDTO> searchCar(SearchCarDTO searchCarDTO);

    List<BidDTO> getBids();

    boolean changeBidStatus(Long bidId, String status);
}
